version: '3'

services:
    mysql:
        image: mysql:5.7
        platform: linux/x86_64
        environment:
            MYSQL_ROOT_PASSWORD: '${DB_PASSWORD:-root}'
            MYSQL_ROOT_HOST: "%"
            MYSQL_DATABASE: '${DB_DATABASE:-test}'
            MYSQL_USER: '${DB_USERNAME:-dev_user}'
            MYSQL_PASSWORD: '${DB_PASSWORD:-pass}'
            MYSQL_ALLOW_EMPTY_PASSWORD: 1
            TZ: 'UTC'
        volumes:
            - 'mysql:/var/lib/mysql'
            - ./mysql/conf/my.cnf:/etc/mysql/my.cnf
        networks:
            - docker-dev-env
        healthcheck:
            test: [ "CMD", "mysqladmin", "ping", "-p${DB_PASSWORD:-root}" ]
            retries: 3
            timeout: 5s

    phpmyadmin-dev:
        image: phpmyadmin:5.2-apache
        networks:
            - docker-dev-env
        links:
            - mysql
        environment:
            PMA_HOST: mysql
            PMA_PORT: 3306
            PMA_ARBITRARY: 1
        restart: always
        ports:
            - "8081:80"

volumes:
    mysql:
        driver: local

networks:
    docker-dev-env:
        driver: bridge
